### DISCUSSION TITLE
Super-fast build on Mac M1 Pro chip

### DISCUSSION THREAD

**Discussion by grege2** [eteran/nedit-ng#333] [General]
I thought that headline would be more interesting :=)

Thanks @anjohnson for the Wiki info on your Apple M1 build, and the "all in one" Makefile.

I have also got myself a Macbook Pro M1 Pro (too many Pro's in there). This was rather forced on me as I had my 8 year-old Macbook Pro stolen while on holiday a few weeks ago. (Which was working -perfectly- 
 well, I had little intention of upgrading it. Ok, but it wouldn't upgrade to Monterey, so the time was coming).

I fired up Andrew's makefile and it flew through the build. That "make -j8" was really working. Comparison times:

```
$ time make -j1
...
[100%] Built target nedit-import
real	1m43.345s
user	1m42.130s
sys	0m10.326s

$ time make -j8
...
[100%] Built target nedit-ng
real	0m20.171s
user	1m56.921s
sys	0m11.011s
```
So, 5 times faster. It's cool to have 8 cores. Actually there are 10, but 2 are "efficiency cores", I suppose they don't fire up for this build. I tried ``make -j12`` but no improvement. 

Back to more relevant info - the build looks fine. I'm using the Master branch which seems to be dated 05Apr22. Build is on Monterey 12.4. Revs of tools are similar to Andrew's list, a couple of tiny steps more recent. I'll put a brief note on the Wiki. 

Ciao tutti.

**Comment by eteran** [comment]
Oh yes, using `-j` will yield a MASSIVE speedup. Glad that it helped you so much!

Side note, I often find myself using the `ninja` build system since it automatically decides to use multicore builds with no user intervention. It usually looks somewhat like this:

```
$ cmake -GNinja ..
$ ninja
```

assuming you have the ninja build system installed of course.
